opal/mca/memory/base/memory_base_open.c

```diff
@@ -65,31 +69,64 @@ static opal_memory_base_component_2_0_0_t empty_component = {
  */
 opal_memory_base_component_2_0_0_t *opal_memory = &empty_component;
 
+#if MEMORY_LINUX_PTMALLOC2
+#include "opal/mca/memory/linux/memory_linux.h"
+#endif
 
+void opal_memory_base_malloc_init_hook (void)
+{
+#if MEMORY_LINUX_PTMALLOC2
```

Comment:
I find it a bit weird to have #if's for specific components in the base... is this really necessary?